File: c/struct/B-AAC.c
// This is similar to the Rust impl, see in the file B-AAC.rs

#include <stdio.h>

typedef struct{
    char name[40];
    int age;
    void (*sayHi)(void);
    
} person;

void sayHi(){
    printf("Hi");
}


void main() {

    person nuni; 
        strcpy(nuni.name, "nuni"); 
        nuni.age = 20;

    nuni.sayHi();
}

File: c/struct/B-AAC.rs

#[warn(dead_code)]
struct Person{
    name: String,
    age: u8,
}

impl Person {
    fn say_hi(&self){
        println!("Hi");
    }
}


fn main() {
    let nuni = Person {
        name: String::from("nuni"),
        age: 20,
    };

    nuni.say_hi();
}

